packages/block-library/src/heading/style.scss

```diff
@@ -21,3 +22,27 @@ h6 {
 		padding: $block-bg-padding--v $block-bg-padding--h;
 	}
 }
+
+:root h1 {
```

Comment:
In the [global styles UI](https://github.com/WordPress/gutenberg/issues/21030), we aim to expose base and scale, and not font size directly. At the block UI level, we're instead inlining font-size values directly (see [related PR](https://github.com/WordPress/gutenberg/pull/21153) to make font size an implicit attribute of the block). This may need some consolidation in further PRs.

We also may do with some pre-generation of variables or/and with some improvements to the theme.json, as @jorgefilipecosta ([suggested](https://github.com/WordPress/gutenberg/pull/20530#discussion_r391767806)). I think we may want to explore that separately.

Reply:
Can you share a little clarification on this? Currently the styles here auto-generate heading sizes, and do so with the `:root h1 {}` specificity, which overrides all of the usual theme-provided `h1 {}` heading styles. Is that the intended approach? I'm a little unclear how themes are supposed to provide heading sizes at the moment.

Reply:
I'm so incredibly excited about these features to land, and some of my favorite people are working on this! 

However, and especially seeing the GIF above, the "base font" and "base scale" is the only aspect that feels unintuitive to me. I understand that the idea is to provide a harmonious typescale, and to provide a single control to affect all headings in one go. 

But my first question is "how do I adjust only the H2?"

Tying this to a type-scale only feels like it might end up frustrating users. What if I want my H2 to be almost exactly as big as my H1, and my H3 and H4 to be much smaller? If I'm reading this interface correctly, that's not possible in a global sense. More often than not the designs I've worked with use heading sizes that don't fit a typescale at all, but are still appropriate for the design in question. Many times they even use different font-weights or even fonts at different levels.

Don't get me wrong, the type scale is impressive, and there are many reasons to use one. But to me the type scale feels like an _option_ to offer, rather than being the default or only interface. 

CC: @karmatosed

Reply:
There are a few things at play here:

- Design-wise, I don't think I can personally provide any more input than what it's being discussed at https://github.com/WordPress/gutenberg/issues/21030#issuecomment-606675549 The current direction is that users & themes should set base & scale. Blocks & Themes should use those variables to set the font-size.

- There is also the `:root h*` bit, I can comment more on this. We're experimenting with adding some CSS variables to blocks by default (more context in [thread](https://github.com/WordPress/gutenberg/pull/21037#discussion_r397068844)). Some examples of current merged changes are line-height for [heading](https://github.com/WordPress/gutenberg/pull/21215/files#diff-f2f5efb13e0e16e61312aa9b0c028ae1R1), color & background-color for [paragraph](https://github.com/WordPress/gutenberg/pull/21037/files#diff-6aca82de6dd8ae08de9708ae9ed73ce6R1), [heading](https://github.com/WordPress/gutenberg/pull/21039/files#diff-f2f5efb13e0e16e61312aa9b0c028ae1R6), and [Media/Text](https://github.com/WordPress/gutenberg/pull/21169/files#diff-eb63570fda7a811f34f5ce8f58ed4d37R1). This PR expands on that approach and adds font-weight and font-size by default to paragraph & heading. The vision is that this allows themes to config the CSS variables via the `theme.json` without needing to use them directly unless they explicitly want to expand what core does or do things differently.

Reply:
Lots of great conversation here. I wanted to pop in and first say nothing here is set. I think what was mocked up was a good first bet, let's maybe see where this needs iterating.

There are a few options:

- Just a scale: this has limited flexibility, however it could use the body text size to build from.
- Set a base: this feels a more flexible approach but it also has complications.

I can totally see maybe this is theme optional, maybe it all is for base font and type scale. I however, do see that a user could find this really magical being able to go up and down.

Regarding individual headings, this is where I agree there needs to be some considered refinement. We are considering elements in groups, over individually. One option which is debatable if v1 or beyond (I could go either way), could be to have an option for more granular control 'reveal'. A simple then list of each heading could do that for headings. I'm totally into exploring around this.